### PR TITLE
[containerapp] fix sessionpool registryServer to Server and fix appending of secrets

### PR DESCRIPTION
**Pull request by smzeng**
---
Bug: 
bug handling ‘customContainerTemplate.registryCredentials.registryServer' Unmatched property name. 
Fix: 
change registryServer to server

Bug:
When passing --registry-password with password directly, not in secretRef way, the secret generated not save in the payload property.

Fix:
after setting up registry credential pass the updated secrets list back and update the secrets definition to the updated secrets definition in the main method.
 
This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
containerapp

**Review by yalixiang — APPROVED** [review]

**Comment by yanzhudd** [comment]
please fix the code conflict.

File: src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py [hunk 1]
```diff
@@ -415,17 +416,32 @@ def set_up_container(self, customer_container_template):
...
         if self.get_argument_registry_user() is not None:
             safe_set(customer_container_template, "registryCredentials", "username", value=self.get_argument_registry_user())
         if secrets_def is None:
             secrets_def = []
         if self.get_argument_registry_pass() is not None:
+            original_secrets = self.existing_pool_def["properties"]["secrets"]
+            original_secrets_names = []
+            for secret in original_secrets:
+                original_secrets_names.append(secret["name"])
+            logger.warning(original_secrets)
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py:429]: We should remove this `logger.warning(original_secrets)`, right?

Comment (yalixiang) [reply src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py]: Updated.

**Review by Greedygre — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**